projpicker/web/web.py

```diff
@@ -0,0 +1,177 @@
+import webbrowser
+import argparse
+
+import projpicker as ppik
+
+
+class Color:
+    HEADER = "\033[95m"
+    OKBLUE = "\033[94m"
+    OKCYAN = "\033[96m"
+    OKGREEN = "\033[92m"
+    WARNING = "\033[93m"
+    FAIL = "\033[91m"
+    ENDC = "\033[0m"
+    BOLD = "\033[1m"
+    UNDERLINE = "\033[4m"
+
+
+class Geometry:
+    def __init__(self, typ, coors):
+        self.type = "poly" if typ == "Polygon" else "point"
+
+        # Reverse coordinates as leaflet returns opposite order of what
+        # ProjPicker takes
+        if self.type == "point":
+            # Coordinates in "Point" type are single-depth tuple [i, j]
+            self.coors = coors[::-1]
+        else:
+            # Coordinates in "Poly" type are in multi-depth array of size
+            # [[[i0, j0], [i1, j1], ...]]; Move down array depth for easier
+            # iteration
+            latlon_coors = []
+            for lonlat in coors[0]:
+                latlon_coors.append(lonlat[::-1])
+            self.coors = list(latlon_coors)
+
+
+class HTTPRequestHandler(http.server.SimpleHTTPRequestHandler):
+    global projpicker_query
+
+    def message(self, *args):
+        if hasattr(self.server, "message"):
+            self.server.message(*args)
+        else:
+            print(*args)
+
+    def do_GET(self):
+        if self.path == "/":
+            self.path = "index.html"
+            return http.server.SimpleHTTPRequestHandler.do_GET(self)
+
+        elif self.path == "/ppikdata":
+            self.send_response(200)
+            self.send_header("Content-type", "text/html")
+            self.end_headers()
+            with open("projdata", "r") as f:
+                self.wfile.write(bytes(f.read(), "utf8"))
+
+    def do_POST(self):
+        if self.path == "/data":
+            # http.client.HTTPResponse stores headers and implements
+            # email.message.Message class'
+            # https://docs.python.org/3/library/email.compat32-message.html#email.message.Message
+            header = self.headers
+
+            content_len = int(header.get("content-length"))
+            content_charset = header.get_content_charset(failobj="utf-8")
+            content_bytes = self.rfile.read(content_len)
+            content_body = content_bytes.decode(content_charset)
+            projpicker_query = content_body
+
+            geoms = create_parsable_geoms(json.loads(content_body))
+
+            self.message(f"{Color.BOLD}ProjPicker query{Color.ENDC}")
+            self.message(f"{Color.BOLD}{'-'*79}{Color.ENDC}")
+            self.message(geoms)
+            self.message(f"{Color.BOLD}{'-'*79}{Color.ENDC}")
+
+            geoms = ppik.parse_mixed_geoms(geoms)
+            self.message(f"Query geometries: {geoms}")
+
+            crs_list = ppik.query_mixed_geoms(geoms)
+            self.message(f"Number of queried CRSs: {len(crs_list)}")
+
+            projpicker_query = bbox_to_json(crs_list)
+
+            projpicker_output_json = open("ppikdata", "w")
```

Comment:
This is not multi-clients safe. We need to send this data directly to the client instead of writing it on the disk. Another user can pick it up (either partially written or someone else's data).

Reply:
Now directly send data to the client in https://github.com/HuidaeCho/projpicker/pull/40/commits/209162b1aef631bea4e3cc70d4d0d699558028f2